opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,724 @@
+        Return:
+             None
+        """
+        if model_id is None:
+            self.model_id = "sentence-transformers/msmarco-distilbert-base-tas-b"
+        else:
+            self.model_id = model_id
+
+    def train(
+        self,
+        read_path: str,
+        output_model_name: str = None,
+        output_model_path: str = None,
+        zip_file_name: str = None,
+        use_accelerate: bool = False,
+        compute_environment: str = None,
+        num_machines: int = 1,
+        num_processes: int = None,
+        learning_rate: float = 2e-5,
+        num_epochs: int = 20,
+        verbose: bool = False,
+    ) -> None:
+        """
+        Description:
+        Read the synthetic queries and use it to fine tune/train (and save) a sentence transformer model.
+
+        Parameters:
+        read_path: str
+            required, path to the zipped file that contains generated queries, if None, raise exception.
+            the zipped file should contain pickled file in list of dictionary format with key named as 'query',
+            'probability' and 'passages'. For example: [{'query':q1,'probability': p1,'passages': pa1}, ...].
+            'probability' is not required for training purpose.
+        output_model_path: str=None
+            the path to store trained custom model. If None, default as current folder path
+        output_model_name: str=None
+            the name of the trained custom model. If None, default as 'trained_model.pt'
+        zip_file_name: str =None
+            Optional, file name for zip file. if None, default as custom_tasb_model.zip
+        use_accelerate: bool = False,
+            Optional, use accelerate to fine tune model. Default as false to not use accelerator to fine tune model.
+            If there are multiple gpus available in the machine, it's recommended to use accelerate with num_processor>1
+            to speeed up the training progress.
+        compute_environment: str
+            optional, compute environment type to run model, if None, default using 'LOCAL_MACHINE'
+        num_machines: int
+            optional, number of machine to run model , if None, default using 1
+        num_processes: int
+            optional, number of processors to run model , if None, default using 1
+        learning_rate: float
+            optional, learning rate to train model, default is 2e-5
+        num_epochs: int
+            optional, number of epochs to train model, default is 20
+        verbose: float
+            optional, use plotting to plot the training progress. Default as false.
+        Return:
+            None
+        """
+
+        query_df = self.read_queries(read_path)
+
+        train_examples = self.load_sentence_transformer_example(
+            query_df, use_accelerate
+        )
+
+        # if use accelerator to train model, run auto setup accelerate confi and launch train_model function
```

Comment:
What do you think putting these comments in the `docString` section of the function?